gh-79516: allow msgfmt.py to compile multiple input po files (#10875)
python/cpython: Misc/NEWS.d/next/Tools-Demos/2018-12-05-20-46-10.bpo-35335.qtIUBx.rst

> :program:`msgfmt.py` is now able to merge more than one single po file into a compiled mo

Reviewer: ```suggestion
:program:`msgfmt.py` is now able to merge more than one po file into a compiled mo
```
Repetition

> The script ``Tools/i18n/msgfmt.py`` is now able to  merge more than one single po file into a compiled mo

Reviewer: ```suggestion
:program:`msgfmt.py` is now able to  merge more than one single po file into a compiled mo
```

@tomasr8 :-)

> The script ``Tools/i18n/msgfmt.py`` is now able to  merge more than one single po file into a compiled mo

Reviewer: ```suggestion
The script ``Tools/i18n/msgfmt.py`` is now able to merge more than one single po file into a compiled mo
```
inconsistent spacing

file. When an entry exists in more than on input file, the last file wins.

> It is also possible to directly call ``make`` more than once

Reviewer 2: This entry should mention that it’s about the msgfmt.py tool!

I would delete this last line: internal detail

Author: The ability to directly call make after importing the file is not a detail IMO. I depend on it for another project of mine, a hatchling plugin that compiles PO files to MO ones when building a wheel.
For the *image* point, I may have use a poor translation from French (English is not my first language...). I meant a copy of the binary content. Could you suggest me a better word?

Author: I hope my last commits have addressed your remarks.

Reviewer 2: I understand what you’re doing, but I still don’t think it’s part of the advertised and supported interface for msgfmt.py. It’s not a stdlib module, just a script that’s not even installed by most distributions of Python.

For images, if you meant _captures à un moment donné_ or _instantanés_, the word would be snapshot! (je suis aussi francophone 🥖)

Reviewer 2: `contents` like you used is good!

Reviewer 3: I agree with @merwok that we shouldn't make any guarantees about the internals of msgfmt - it's a script and (for now anyway) lives in `Tools`. We actually recently added a note clarifying this to the help output of pygettext:

https://github.com/python/cpython/blob/c9932a9ec8a3077933a85101aae9c3ac155e6d04/Tools/i18n/pygettext.py#L38-L39

Maybe we should do the same with msgfmt?

Author: I can understand. In another project of mine, I already import that *modified* version of `msgfmt.py` and call its make function. That is the reason why I wanted to allow it and added a test for that. I now realize that it is nonsense to declare a public API in a script from Tools in the main CPython source. I shall follow your advice and cleanup the comments and tests here, and keep my original code in my other project as a *derivative work based on CPython source code* keeping the Python Software Foundation License Version 2
